stdlib/2/sets.pyi

```diff
@@ -0,0 +1,60 @@
+# Stubs for sets (Python 2)
+from typing import Any, Callable, Hashable, Iterable, Iterator, Mapping, Optional, TypeVar, Union
+
+_T = TypeVar('_T')
+Setlike = Union['BaseSet', Iterable]
+
+class BaseSet(Iterable[_T]):
+    def __init__(self) -> None: ...
+    def __len__(self) -> int: ...
+    def __repr__(self) -> str: ...
+    __str__ = ...  # type: Callable[[BaseSet[_T]], str]
+    def __iter__(self) -> Iterator[_T]: ...
+    def __cmp__(self, other: Any) -> None: ...
+    def __eq__(self, other: Any) -> bool: ...
+    def __ne__(self, other: Any) -> bool: ...
+    def copy(self) -> BaseSet[_T]: ...
+    __copy__ = ...  # type: Callable[[BaseSet[_T]], BaseSet[_T]]
+    def __deepcopy__(self, memo: Mapping[int, BaseSet[_T]]) -> BaseSet[_T]: ...
+    def __or__(self, other: BaseSet[_T]) -> BaseSet[_T]: ...
+    def union(self, other: Setlike) -> BaseSet[_T]: ...
+    def __and__(self, other: BaseSet[_T]) -> BaseSet[_T]: ...
+    def intersection(self, other: Setlike) -> BaseSet[_T]: ...
+    def __xor__(self, other: BaseSet[_T]) -> BaseSet[_T]: ...
+    def symmetric_difference(self, other: Setlike) -> BaseSet[_T]: ...
+    def __sub__(self, other: BaseSet[_T]) -> BaseSet[_T]: ...
+    def difference(self, other: Setlike) -> BaseSet[_T]: ...
+    def __contains__(self, element: Any) -> bool: ...
+    def issubset(self, other: BaseSet[_T]) -> bool: ...
+    def issuperset(self, other: BaseSet[_T]) -> bool: ...
+    __le__ = ...  # type: Callable[[BaseSet[_T], BaseSet[_T]], bool]
+    __ge__ = ...  # type: Callable[[BaseSet[_T], BaseSet[_T]], bool]
+    def __lt__(self, other: BaseSet[_T]) -> bool: ...
+    def __gt__(self, other: BaseSet[_T]) -> bool: ...
+
+class ImmutableSet(BaseSet[_T], Hashable):
+    def __init__(self, iterable: Optional[Setlike] = ...) -> None: ...
```

Comment:
Note that _Setlike is not a TypeVar, but instead an alias for the Union of BaseSet[_T] and Iterable[_T].